### ISSUE TITLE
Formatter - PercentageFormat - ArgumentCountError: 4 arguments are required, 2 given

### ISSUE THREAD

**Issue by FlameStorm** [PHPOffice/PhpSpreadsheet#2885]
This is a bug report according to reading loaded XLSX file raising a Fatal error in sprintf due to incorrect format string.

### What is the expected behavior?
Just normal working with loaded XLSX files.

### What is the current behavior?
Got a Fatal error during reading loaded XLSX file:

ArgumentCountError: 4 arguments are required, 2 given

/var/www/vendor/phpoffice/phpspreadsheet/src/PhpSpreadsheet/Style/NumberFormat/PercentageFormatter.php:43
/var/www/vendor/phpoffice/phpspreadsheet/src/PhpSpreadsheet/Style/NumberFormat/Formatter.php:146
/var/www/vendor/phpoffice/phpspreadsheet/src/PhpSpreadsheet/Style/NumberFormat.php:404
/var/www/vendor/phpoffice/phpspreadsheet/src/PhpSpreadsheet/Worksheet/Worksheet.php:2835

### What are the steps to reproduce?

Please provide a [Minimal, Complete, and Verifiable example](https://stackoverflow.com/help/mcve) of code that exhibits the issue without relying on an external Excel file or a web server:

```php
<?php

require __DIR__ . '/vendor/autoload.php';

// Create new Spreadsheet object
$reader = \PhpOffice\PhpSpreadsheet\IOFactory::createReader("Xlsx");
$spreadsheet = $reader->load("./percents_test_case.xlsx");
$sheet = $spreadsheet->getActiveSheet();
$finishColumn = $sheet->getHighestColumn();
$rowsCount = $sheet->getHighestRow();
$rows = $sheet->rangeToArray("A1:{$finishColumn}{$rowsCount}");
// ^ Fatal error

```

The original (and external for us, we can't do anything with it's generation) loaded XLSX file contains format like `<numFmt numFmtId="84" formatCode="[$-010419]0%">` in `<numFmts ...>` block at /xl/styles.xml ("under zip"). 
Such custom/special formats with prepending `[$-...]` in formatCode are allowed as described in documentation for `numberingformat` at https://docs.microsoft.com/en-us/dotnet/api/documentformat.openxml.spreadsheet.numberingformat?view=openxml-2.8.1 and really can be set in MS Excel, LibreOffice Calc or another spreadsheet editor.

Sample file for showing the case:
[percents_test_case.xlsx](https://github.com/PHPOffice/PhpSpreadsheet/files/8899312/percents_test_case.xlsx)

### What features do you think are causing the issue

- [x] Reader
- [ ] Writer
- [x] Styles
- [ ] Data Validations
- [ ] Formula Calulations
- [ ] Charts
- [ ] AutoFilter
- [ ] Form Elements

### Does an issue affect all spreadsheet file formats? If not, which formats are affected?
It affects some subset of XLSX files.

### Which versions of PhpSpreadsheet and PHP are affected?
Latest, phpoffice/phpspreadsheet 1.23.0

**Comment by FlameStorm** [comment]
A bit of illustrations:

Some cell formatting processing:

Into NumberFormat.php :

![image](https://user-images.githubusercontent.com/1779603/173576779-f16de01a-e278-48a7-bcf2-67bcaf7e0abf.png)

![image](https://user-images.githubusercontent.com/1779603/173577029-ddea21b6-dba9-4ea6-aac0-31fbb70b775f.png)

Into PercentageFormatter.php :

![image](https://user-images.githubusercontent.com/1779603/173577314-099ebd64-f721-4721-a562-317b8d68930b.png)

(bug is coming!)
![image](https://user-images.githubusercontent.com/1779603/173577444-d731b50b-82c8-469a-9671-256a109519df.png)

Source XLSX file's `xl/styles.xml` content:

![image](https://user-images.githubusercontent.com/1779603/173577751-442847b7-b4f0-40d6-93c7-7c5a3f57ccfb.png)

**Comment by FlameStorm** [comment]
PR https://github.com/PHPOffice/PhpSpreadsheet/pull/2886 was updated with respect of given feedback.

**Comment by oleibman** [comment]
This issue should be resolved by PR #2886, which was just merged.
